sunpy/io/special/asdf/tags/map/generic_map.py

```diff
@@ -19,7 +18,8 @@ class GenericMapType(SunPyType):
     def from_tree(cls, node, ctx):
         # Use the factory here to get the correct subclass back
         out_map = sunpy.map.Map(np.asarray(node['data']), node['meta'])
-        out_map.shift(*node['shift'])
+        if 'shift' in node:
+            out_map = out_map.shift_reference_coord(*node['shift'])
```

Comment:
Ideally we would test this by making a file before this change with a shift and asserting that the shift gets applied after it.

Reply:
I would argue that's not in scope for this PR.

Reply:
I did it. Code without tests shall not pass :p